docs/AQL/grammar/AqlLexer.g4

```diff
@@ -0,0 +1,207 @@
+//
+//  description:  ANTLR4 lexer grammar for Archetype Query Language (AQL)
+//  author:       Sebastian Iancu
+//  contributors: This version of the grammar is a complet rewrite of previously published antlr3 grammar,
+//                based on current AQL specifications in combination with other grammars from several AQL implementations.
+//                The openEHR Foundation would like to recognise the following people for their contributions:
+//                  - Chunlan Ma & Heath Frankel, Ocen Health Systems, Australia
+//                  - Bostjan Lah, Better, Slovenia
+//                  - Christian Chevalley, EHRBase, Germany
+//                  - Teun van Helmert, Nedap, Netherlands
+//  support:      openEHR Specifications PR tracker <https://specifications.openehr.org/releases/QUERY/open_issues>
+//  copyright:    Copyright (c) 2020 openEHR Foundation
+//  license:      Apache 2.0 License <http://www.apache.org/licenses/LICENSE-2.0.html>
+//
+
+lexer grammar AqlLexer;
+
+channels {
+    COMMENT_CHANNEL
+}
+
+// SKIP
+WS: [ \t\r\n]+ -> skip;
+UNICODE_BOM: (
+    '\uEFBBBF' // UTF-8 BOM
+    | '\uFEFF' // UTF16_BOM
+    | '\u0000FEFF' // UTF32_BOM
+    ) -> skip;
+COMMENT: (
+    MINUSMINUS ' ' ~[\r\n]* ('\r'? '\n' | EOF)
+    | MINUSMINUS ('\r'? '\n' | EOF)
+    ) -> channel(COMMENT_CHANNEL);
+
+// Keywords
+// Common Keywords
+SELECT: S E L E C T ;
+AS: A S ;
+FROM: F R O M ;
+WHERE: W H E R E ;
+ORDER: O R D E R ;
+BY: B Y ;
+DESC: D E S C ;
+DESCENDING: D E S C E N D I N G ;
+ASC: A S C ;
+ASCENDING: A S C E N D I N G ;
+LIMIT: L I M I T ;
+OFFSET: O F F S E T ;
+// deprecated
+TOP: T O P ;
+FORWARD: F O R W A R D ;
+BACKWARD: B A C K W A R D ;
+
+// Operators
+// Containment operator
+CONTAINS : C O N T A I N S ;
+// Logical operators
+AND : A N D ;
+OR : O R ;
+NOT : N O T ;
+EXISTS: E X I S T S ;
+// Arithmetical operators
+ARITHMETIC_OPERATOR: '*' | '/' | '+' | '-';
+// Comparison operators
+COMPARISON_OPERATOR: '=' | '!=' | '>' | '>=' | '<' | '<=' ;
+LIKE: L I K E ;
+MATCHES: M A T C H E S ;
+
+// functions
```

Comment:
The specification at some point also mentions NOW, CURRENT-DATE and CURRENT-DATE-TIME, but these are not currently supported in either the lexer or the grammar. I don't know if this is intentional?